internal/library_sync_machine.go

```diff
@@ -0,0 +1,198 @@
+package internal
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-pg/pg/v10"
+	"github.com/zhammer/stride-songs/pkg/spotify"
+)
+
+type librarySyncStatus string
+
+const (
+	LibrarySyncStatusPendingRefreshToken = librarySyncStatus("pending_refresh_token")
+	LibrarySyncStatusCreatingPlaylists   = librarySyncStatus("creating_playlists")
+	LibrarySyncStatusScanningLibrary     = librarySyncStatus("scanning_library")
+	LibrarySyncStatusAddingTracks        = librarySyncStatus("adding_tracks")
+	LibrarySyncStatusSucceeded           = librarySyncStatus("succeeded")
+)
+
+type LibrarySyncMachine struct {
+	*StrideSongs
+}
+
+func (sm *LibrarySyncMachine) HandleStateUpdate(ctx context.Context, old User, new User) error {
```

Comment:
I wish there were a clearer mapping of state transitions. This is just one after another so not so bad, but still would like to be able to see "adding_tracks" -> succeeded